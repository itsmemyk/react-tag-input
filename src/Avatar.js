import React, { Component, PropTypes } from 'react';

export const matchNames = ({ username, fullName }, keyword) => {
	keyword = keyword.toLowerCase();
	fullName = fullName.toLowerCase();
	username = username.toLowerCase();
	const splits = fullName.split(' ');
	return !keyword || (username.indexOf(keyword) > -1 || fullName.indexOf(keyword) > -1 || keyword === splits[0][0] + '' + splits[1][0]);
}

export class Avatar extends Component {
	render() {
		const { thumb, username, fullName } = this.props;
		return(
			<div>
				{
					thumb &&
					<span> <img src={thumb} alt={fullName} /> </span>
				}
				<b> {username} </b>
				<span> {fullName} </span>
			</div>
		);
	}
}

Avatar.propTypes = {
	username: PropTypes.string,
	thumb: PropTypes.string,
	fullName: PropTypes.string,
};

export default Avatar;
