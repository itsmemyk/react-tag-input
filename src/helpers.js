export function queryAllSelector(element, selector, value) {
	let results = [];

	if (selector(element, value)) {
		results.push(element);
	}

	if (element.children) {
		element.children.forEach((child) => {
			results = results.concat(queryAllSelector(child, selector, value))
		});
	}
	return results;
}

export function findAllByTag(element, tagName) {
	return queryAllSelector(
		element,
		(element, value) => element.type === value,
		tagName,
	);
}

export function findAllByClassName(element, className) {
	return queryAllSelector(
		element,
		(element, value) => element.props && element.props.className && element.props.className === value,
		className,
	);
}

export function findByTag(element, tagName) {
	return (findAllByTag(element, tagName) || [null])[0];
}

export function findByClassName(element, className) {
	return (findAllByClassName(element, className) || [null])[0];
}
