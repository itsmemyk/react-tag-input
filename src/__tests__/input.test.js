import React from 'react';
import renderer from 'react-test-renderer';

import { Avatar as RenderAvatar, matchNames } from '../Avatar';
import InputControl from '../input';
import { findByTag } from '../helpers';

function createNodeMock(element) {
  if (element.type === 'input') {
    return {
      focus() {},
			selectionStart: 0,
    };
  }
  return null;
}

const peoples = [
	{ thumb: 'https://ca.slack-edge.com/T0S9307C3-U0TBEUYUF-ga0b7c75171c-48', username: 'itsmemyk', fullName: 'Mayank Mahadevwala' },
	{ thumb: 'https://ca.slack-edge.com/T0S9307C3-U0T3Y7YE9-g741fb107a95-48', username: 'sos', fullName: 'Soham Solanki' },
	{ thumb: 'https://ca.slack-edge.com/T0S9307C3-USLACKBOT-sv1444671949-48', username: 'tirupa', fullName: 'Tirupa Magroliya' },
	{ thumb: 'https://ca.slack-edge.com/T0S9307C3-U0SS32SG7-g4c8f334018e-48', username: 'amit', fullName: 'Amit Patel' },
	{ thumb: 'https://ca.slack-edge.com/T0S9307C3-U10AWP6A1-g6c660b98934-48', username: 'bhavikkheni', fullName: 'Bhavik Kheni' },
	{ username: 'everyone', fullName: 'Notify everyone in your team' },
	{ username: 'here', fullName: 'Notify everyone in your team' },
];

describe('Featured Input Component', () => {
	let app, tree, onChange;
	let snap = () => {
		return tree = app.toJSON();
	}

	beforeEach(() => {
		app = renderer.create(
				<InputControl
					onChange={onChange}
					tagSymbol="@"
					data={peoples}
					predicate={{
						key: 'username',
						selector: matchNames,
						element: <RenderAvatar />,
					}}
				/>
		, { createNodeMock });
	});

	it('Should render correctly', () => {
		expect(snap()).toMatchSnapshot();
	});


	it('Should show popup on tag char i.e. @', () => {
		expect(snap()).toMatchSnapshot();
		let input = findByTag(tree, 'input');
		input.props.onChange({ target: { value: '@' }});
		expect(snap()).toMatchSnapshot();
	});
});