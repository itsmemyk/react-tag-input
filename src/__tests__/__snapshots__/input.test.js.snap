exports[`Featured Input Component Should render correctly 1`] = `
<div
  className="custom-input-control">
  <div
    className="form-group">
    <span
      className="input-group">
      <span
        className="input-group-addon">
         + 
      </span>
      <input
        className="form-control"
        id={undefined}
        onChange={[Function]}
        onKeyDown={[Function]}
        type="text"
        value="" />
    </span>
  </div>
  <div
    className="custom-popup" />
</div>
`;

exports[`Featured Input Component Should show popup on tag char i.e. @ 1`] = `
<div
  className="custom-input-control">
  <div
    className="form-group">
    <span
      className="input-group">
      <span
        className="input-group-addon">
         + 
      </span>
      <input
        className="form-control"
        id={undefined}
        onChange={[Function]}
        onKeyDown={[Function]}
        type="text"
        value="" />
    </span>
  </div>
  <div
    className="custom-popup" />
</div>
`;

exports[`Featured Input Component Should show popup on tag char i.e. @ 2`] = `
<div
  className="custom-input-control">
  <div
    className="form-group">
    <span
      className="input-group">
      <span
        className="input-group-addon">
         + 
      </span>
      <input
        className="form-control"
        id={undefined}
        onChange={[Function]}
        onKeyDown={[Function]}
        type="text"
        value="@" />
    </span>
  </div>
  <div
    className="custom-popup">
    <div
      className="panel panel-default"
      id={undefined}>
      <div
        className="panel-heading">
        <div
          className="panel-title">
          People 
        </div>
      </div>
      <div
        className="panel-body">
        <ul>
          <li
            className="selected"
            onClick={[Function]}
            onMouseEnter={[Function]}>
            <div>
              <span>
                 
                <img
                  alt="Mayank Mahadevwala"
                  src="https://ca.slack-edge.com/T0S9307C3-U0TBEUYUF-ga0b7c75171c-48" />
                 
              </span>
              <b>
                 
                itsmemyk
                 
              </b>
              <span>
                 
                Mayank Mahadevwala
                 
              </span>
            </div>
          </li>
          <li
            className=""
            onClick={[Function]}
            onMouseEnter={[Function]}>
            <div>
              <span>
                 
                <img
                  alt="Soham Solanki"
                  src="https://ca.slack-edge.com/T0S9307C3-U0T3Y7YE9-g741fb107a95-48" />
                 
              </span>
              <b>
                 
                sos
                 
              </b>
              <span>
                 
                Soham Solanki
                 
              </span>
            </div>
          </li>
          <li
            className=""
            onClick={[Function]}
            onMouseEnter={[Function]}>
            <div>
              <span>
                 
                <img
                  alt="Tirupa Magroliya"
                  src="https://ca.slack-edge.com/T0S9307C3-USLACKBOT-sv1444671949-48" />
                 
              </span>
              <b>
                 
                tirupa
                 
              </b>
              <span>
                 
                Tirupa Magroliya
                 
              </span>
            </div>
          </li>
          <li
            className=""
            onClick={[Function]}
            onMouseEnter={[Function]}>
            <div>
              <span>
                 
                <img
                  alt="Amit Patel"
                  src="https://ca.slack-edge.com/T0S9307C3-U0SS32SG7-g4c8f334018e-48" />
                 
              </span>
              <b>
                 
                amit
                 
              </b>
              <span>
                 
                Amit Patel
                 
              </span>
            </div>
          </li>
          <li
            className=""
            onClick={[Function]}
            onMouseEnter={[Function]}>
            <div>
              <span>
                 
                <img
                  alt="Bhavik Kheni"
                  src="https://ca.slack-edge.com/T0S9307C3-U10AWP6A1-g6c660b98934-48" />
                 
              </span>
              <b>
                 
                bhavikkheni
                 
              </b>
              <span>
                 
                Bhavik Kheni
                 
              </span>
            </div>
          </li>
          <li
            className=""
            onClick={[Function]}
            onMouseEnter={[Function]}>
            <div>
              <b>
                 
                everyone
                 
              </b>
              <span>
                 
                Notify everyone in your team
                 
              </span>
            </div>
          </li>
          <li
            className=""
            onClick={[Function]}
            onMouseEnter={[Function]}>
            <div>
              <b>
                 
                here
                 
              </b>
              <span>
                 
                Notify everyone in your team
                 
              </span>
            </div>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
